pkg/trace/info/stats.go

```diff
@@ -155,15 +189,66 @@ func (ts *TagStats) publish() {
 	metrics.Count("datadog.trace_agent.receiver.events_extracted", eventsExtracted, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.events_sampled", eventsSampled, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.payload_accepted", requestsMade, tags, 1)
+
+
+	var droppedReasons []string
+	for _, c := range ts.droppedTraceReasonCounts() {
```

Comment:
This whole thing is a bit too "smart" and slightly confusing. That structure is basically a `map[string]int64`, so let's make it super obvious to a reader what's going on here. This package is the definition of verbose already so let's do the same here:
```go
for reason, count := range map[string]int64{
	"reason:decoding_error": atomic.LoadInt64(&ts.TracesDroppedDecodingError),
	"reason:empty_trace":    atomic.LoadInt64(&ts.TracesDroppedEmptyTrace),
	"reason:trace_id_zero":  atomic.LoadInt64(&ts.TracesDroppedTraceIdZero),
	"reason:span_id_zero":   atomic.LoadInt64(&ts.TracesDroppedSpanIdZero),
	"reason:foreign_span":   atomic.LoadInt64(&ts.TracesDroppedForeignSpan),
} {
	metrics.Count("datadog.trace_agent.normalizer.traces_dropped", count, append(tags, reason), 1)
}
```
...then we don't have to scroll and jump aroind the file to see which struct field belongs to which metric.

Reply:
Yeah it was confusing. I changed it since then, can you take another look?

Reply:
I would prefer it if you used the above suggestion. It significantly reduces the amount of code. I don't think that the extra type and those functions bring any value.

Reply:
Oh I see have another struct `ts.TracesDropped`, yeah that's a bit nicer. Will do

Reply:
> Oh I see have another struct ts.TracesDropped, yeah that's a bit nicer. Will do
That's the discussion in [this thread](https://github.com/DataDog/datadog-agent/pull/3738#discussion_r296568295) 😄